src/js/helpers/graph/declarationHelper.mock.js

```diff
@@ -0,0 +1,374 @@
+      label: 'abcd',
+      status: 'error',
+      type: 'single',
+      on: {
+        _success: {
+          id: 'subtask2-1',
+          label: 'Sub 2-1',
+          status: 'unprocessed',
+          type: 'single',
+        },
+        _error: {
+          id: 'subtask2-2',
+          label: 'Sub 2-2',
+          status: 'error',
+          type: 'single',
+        },
+        custom: {
+          id: 'subtask2-3',
+          label: 'Sub 2-3',
+          status: 'unprocessed',
+          type: 'single',
+        },
+      },
+      response: {
+        transition: '_error',
+      },
+    },
+  ],
+};
+
+export const compositeWithUnevenSubtasks = {
+  id: 'composite',
+  label: 'Abcd',
+  status: 'success',
+  type: 'composite',
+  subtasks: [
+    {
+      id: 'sub1',
+      label: 'Sub 1',
+      status: 'success',
+      type: 'single',
+      on: {
+        _success: {
+          id: 'sub1-1',
+          label: 'Sub 1-1',
+          status: 'success',
+          type: 'single',
+          on: {
+            _success: {
+              id: 'sub1-1-1',
+              label: 'Deepest',
+              status: 'success',
+              type: 'single',
+            },
+          },
+          response: {
+            transition: '_success',
+          },
+        },
+        _fallback: {
+          id: 'sub1-2',
+          label: 'Sub 1-2',
+          status: 'unprocessed',
+          type: 'single',
+        },
+      },
+      response: {
+        transition: '_success',
+      },
+    },
+    {
+      id: 'sub2',
+      label: 'Sub 2',
+      status: 'unprocessed',
+      type: 'single',
+    },
+  ],
+};
+
+export const successNode = {
+  id: 'test',
+  label: 'Test',
+  status: 'success',
+  type: 'single',
+};
+
+export const errorNode = {
+  id: 'test',
+  label: 'Test',
+  status: 'error',
+  type: 'single',
+};
+
+export const missingNode = {
+  id: 'test',
+  label: 'Test',
+  status: 'missing',
+  type: 'single',
+};
+
+export const unprocessedNode = {
+  id: 'test',
+  label: 'Test',
+  status: 'unprocessed',
+  type: 'single',
+};
+
+export const otherNode = {
+  id: 'test',
+  label: 'Test',
+  status: 'other',
+  type: 'single',
+};
+
+export const successCompositeNode = {
+  id: 'test',
+  label: 'Test',
+  status: 'success',
+  type: 'composite',
+  subtasks: [
+    {
+      id: 'subtask',
+      label: 'Subtasl',
+      status: 'success',
+      type: 'single',
+    },
+  ],
+};
+
+export const errorCompositeNode = {
+  id: 'test',
+  label: 'Test',
+  status: 'error',
+  type: 'composite',
+  subtasks: [
+    {
+      id: 'subtask',
+      label: 'Subtasl',
+      status: 'error',
+      type: 'single',
+    },
+  ],
+};
+
+export const unprocessedCompositeNode = {
+  id: 'test',
+  label: 'Test',
+  status: 'unprocessed',
+  type: 'composite',
+  subtasks: [
+    {
+      id: 'subtask',
+      label: 'Subtask',
+      status: 'unprocessed',
+      type: 'single',
+    },
+  ],
+};
+
+export const nestedComposites = {
```

Comment:
Do you check all child nodes? `composite(composite())` should be enough - single test responsibility.